src/node/transaction.cpp

```diff
@@ -50,7 +50,7 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
         // So if the output does exist, then this transaction exists in the chain.
         if (!existingCoin.IsSpent()) return TransactionError::ALREADY_IN_CHAIN;
     }
-    if (!node.mempool->exists(hashTx)) {
+    if (!node.mempool->exists(GenTxid(false, tx->GetHash()))) {
```

Comment:
Is this change altering behavior of `BroadcastTransaction` ? Otherwise maybe better to just rename `hashTx` to `txid`, update comment to "Txid is not already in the mempool. Note if the submitted candidate differs by wtxid from an already present transaction, it's for now rejected", and make it its own commit as it's only a documenting change ?

Reply:
Good point, my intention is merely to highlight/document what the current behavior is. I've now marked the commits with "changes / doesn't change behavior" and switched this to just be a comment explaining the behavior.